mozbattue/main.py

```diff
@@ -118,13 +118,19 @@ def do_trigger(opts):
                 break
     else:
         # we can use the mostly triggered testname as good default
+        # and trigger that for the oldest revision
         most_triggered_bname = \
             intermittents_groupedby_bname(intermittents)[0]['buildname']
-        test_name = split_build_name(most_triggered_bname)['testname']
-        for i in intermittents:
-            if split_build_name(i['buildname'])['testname'] == test_name:
-                oldest = i
-                break
+        try:
+            test_name = split_build_name(most_triggered_bname)['testname']
+            platform_branch = \
+                split_build_name(oldest['buildname'])['platform_branch']
+        except KeyError:
+            # we could have a KeyError if split_build_name fails.
+            # in that case, just build the first intermittent
+            pass
+        else:
+            oldest['buildname'] = '%s test %s' % (platform_branch, test_name)
```

Comment:
is there a concern of reusing the oldest?  could you assign it to a new variable:
target = oldest
try:...
else:
  target['buildername'] = ...

we would need a deepcopy as I believe we could just be passing references around.